modal/_utils/git_utils.py

```diff
@@ -0,0 +1,75 @@
+# Copyright Modal Labs 2025
+import os
+import subprocess
+from typing import Dict, List, Optional, Union
+
+
+def run_command(args: List[str]) -> Optional[str]:
+    """Run a command and return its output.
+
+    Args:
+        args: Command and arguments as a list
+
+    Returns:
+        Command output as string or None if command failed
+    """
+    try:
+        result = subprocess.run(
+            args,
+            check=True,
+            capture_output=True,
+            text=True,
+            shell=False,
+        )
+        return result.stdout.strip()
+    except (subprocess.CalledProcessError, FileNotFoundError):
```

Comment:
Is this every possible error class that could be raised here? IMO we probably never want to block someone running / deploying an App if this fails (but I guess the downside is that we'd never know about a bug...).